src/NUnitFramework/framework/Attributes/RangeAttribute.cs

```diff
@@ -192,12 +202,76 @@ public RangeAttribute(float from, float to, float step)
             float aStep = Math.Abs(step);
             float tol = aStep / 1000;
             int count = (int)(Math.Abs(to - from) / aStep + tol + 1);
-            this.data = new object[count];
+            _data = new object[count];
             int index = 0;
             for (float val = from; index < count; val += step)
-                this.data[index++] = val;
+                _data[index++] = val;
         }
 
         #endregion
+
+        /// <summary>
+        /// Get the range of values to be used as arguments
+        /// </summary>
+        public IEnumerable GetData(IParameterInfo parameter)
+        {
+            Type tartgetType = parameter.ParameterType;
+
+            if (tartgetType.GetTypeInfo().IsEnum && _data.Length == 0)
+            {
```

Comment:
The enum and bool evaluation isn't required for RangeAttribute, is it?

Reply:
Correct. I approved too soon. 😄